### PR TITLE
adding first pass example embeddings logic

### PR DESCRIPTION
**Pull request by NicolasFranklin-MM**
Creating semantic examples branch 

However, the feature is still WIP / incomplete, so thats why this is a draft

### CODE REVIEW

**Comment by NicolasFranklin-MM** [comment]
Not sure where the best place to write some stuff about each commit in more detail, but:
- In commit _**[Examples / List{Example] fix] / d11cc7b:**_ 
Changed the pipe_strategy() method inside pipeline.py to accept "examples" which is a dynamic list of sorted examples by relevance (embedding)
Changed pipeline_prompts.py line about model.json.dump from examples.Examples to simply examples (because now we are casting to List[Example] not Examples
Changed examples.py to cast to List[Example] (I think there were some errors where it said List[Examples] or Examples. Fixed now.

image

File: prompts/pipeline_prompts.py [hunk 1]
```diff
@@ -94,49 +99,94 @@ def format_strategy_examples(examples: list[str]) -> str:
...
-</examples>
-""".format(EXAMPLES=format_strategy_examples(
-    [e.model_dump_json() for e in examples.Examples]
-))
+    <examples>
+    {EXAMPLES}
+    </examples>
+    """.format(EXAMPLES=format_script_examples(
+        [e.model_dump_json() for e in examples]
+        #[e.model_dump_json() for e in examples.Examples]
```
Comment (samgregson) [inline prompts/pipeline_prompts.py:174]: delete zombie code

**Review by samgregson — CHANGES_REQUESTED** [review]
Nice work!

- [x] Please clear the output from the notebooks. I thought I'd implemented an auto clear but doesn't seem to have worked, unless you are missing a push?

- [ ] Could you add a unit test as discussed. i.e. copy what you did for the embeddings notebook into a python file and wrap it into a pytest function, assert that the examples get sorted by checking the first item in the list (e.g. cup example).

**Comment by NicolasFranklin-MM** [comment]
+ Add a test to ensure no embeddings (when doing k-nearest)  

- [ ] Add a test to ensure no embeddings (when doing k-nearest)  - use Assert -> check for e in examples, example.embeddings = None or just check the first one

**Review by samgregson — APPROVED** [review]

**Pull request merged**